portal-kernel/src/com/liferay/portal/kernel/repository/DocumentRepository.java

```diff
@@ -181,6 +187,13 @@ public FileVersion getFileVersion(long fileVersionId)
 	public Folder getFolder(long parentFolderId, String name)
 		throws PortalException;
 
+	public default Folder getFolderByExternalReferenceCode(
+			String externalReferenceCode)
+		throws PortalException {
+
+		return null;
+	}
+
```

Comment:
Q: Shouldn't you throw an exception here? `get*` methods usually throw a `NoSuch*` exception when an element is not found.

Reply:
I supposed that if I throw a `PortalException`, it will be caught by the overrided one. I think that it must be the same as `getFileEntryByExternalReferenceCode` method. You are right!!

Reply:
I will change it!